pennylane/workflow/get_best_diff_method.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a function for getting the best differentiation method for a given QNode.
+
+"""
+
+from functools import wraps
+
+import pennylane as qml
+from pennylane.workflow.qnode import _make_execution_config
+
+
+def get_best_diff_method(qnode):
+    """Returns a function that computes the 'best' differentiation method
+    for a particular QNode.
+
+    This method attempts to determine support for differentiation
+    methods using the following order:
+
+    * ``"device"``
+    * ``"backprop"``
+    * ``"parameter-shift"``
```

Comment:
Can we document what "best" means and why this is the order that we choose? Why is backprop preferred over parameter shift?

Reply:
@astralcai, is the new docstring better?